workload/workloads/userProfileDapi.go

```diff
@@ -0,0 +1,267 @@
+}
+
+func (w userProfileDapi) Probabilities() [][]float64 {
+	return [][]float64{
+		{0, 0.8, 0.1, 0.05, 0.05},
+		{0.6, 0, 0.2, 0.1, 0.1},
+		{0.5, 0.3, 0, 0.15, 0.05},
+		{0.4, 0.3, 0.2, 0, 0.1},
+		{0.3, 0.3, 0.2, 0.2, 0},
+	}
+}
+
+func (w userProfileDapi) Setup() error {
+	// TODO setup FTS index here for findRelatedProfile
+	gofakeit.Seed(int64(workload.RandSeed))
+	return nil
+}
+
+func (w userProfileDapi) Functions() map[string]func(ctx context.Context, rctx workload.Runctx) error {
+	return map[string]func(ctx context.Context, rctx workload.Runctx) error{
+		"fetchProfile":        w.fetchProfile,        // similar to login or looking at someone
+		"updateProfile":       w.updateProfile,       // updating a status on the profile
+		"lockProfile":         w.lockProfile,         // disable or enable a random profile (account lockout)
+		"findProfile":         w.findProfile,         // find a profile by a secondary index (email address)
+		"findRelatedProfiles": w.findRelatedProfiles, // look for people with similar interests
+	}
+}
+
+func (w userProfileDapi) executeRequest(req *http.Request) (*http.Response, error) {
+	req.SetBasicAuth(w.username, w.password)
+	resp, err := w.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute get request: %s", err.Error())
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("profile fetch returned unexpected status code %d", resp.StatusCode)
+	}
+	return resp, nil
+}
+
+// Fetch a random profile in the range of profiles
+func (w userProfileDapi) fetchProfile(ctx context.Context, rctx workload.Runctx) error {
+	id := fmt.Sprintf("u%d", rctx.Rand().Int31n(int32(w.numItems)))
+	requestURL := fmt.Sprintf("%s/v1/buckets/%s/scopes/%s/collections/%s/documents/%s", w.connstr, w.bucket, w.scope, w.collection, id)
+	req, err := http.NewRequest("GET", requestURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to build profile fetch request: %s", err.Error())
+	}
+
+	resp, err := w.executeRequest(req)
+	if err != nil {
+		return fmt.Errorf("could not fetch profile to update: %s", err.Error())
+	}
+
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("could not read response body: %s", err.Error())
+	}
+
+	var toUd User
+	err = json.Unmarshal(bodyText, &toUd)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal response body - %s : %s", string(bodyText), err.Error())
+	}
+	return nil
+}
+
+// Update the status of a random profile
+func (w userProfileDapi) updateProfile(ctx context.Context, rctx workload.Runctx) error {
+	id := fmt.Sprintf("u%d", rctx.Rand().Int31n(int32(w.numItems)))
+	requestURL := fmt.Sprintf("%s/v1/buckets/%s/scopes/%s/collections/%s/documents/%s", w.connstr, w.bucket, w.scope, w.collection, id)
+	req, err := http.NewRequest("GET", requestURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to build profile fetch request: %s", err.Error())
```

Comment:
Same thing.  Open for debate, but anything like this which is basically building a string from local variables should probably panic right there.  Anything that is IO, we should probably try to handle in a way other than whinging to the log.

Reply:
Done!